internal/ui/playlists.go

```diff
@@ -4,29 +4,21 @@ import (
 	"github.com/charmbracelet/bubbles/list"
 	tea "github.com/charmbracelet/bubbletea"
 	"github.com/ehabshaaban/lazyfy/api"
-	"github.com/zmb3/spotify/v2"
 )
 
-var (
-	playlistItems []list.Item
-	state         bool = false
-	foo           TrackModel
-	trackModels   map[spotify.ID]TrackModel = make(map[spotify.ID]TrackModel)
-)
+var trackModels = make(map[string]*trackModel)
 
-type playlistItem struct {
-	id, name string
-}
+type playlist api.Playlist
```

Comment:
This is a bit smelly, why just not use `api.Playlist` directly?

Reply:
When I use  it in title for example: `func (p *api.Playlist) Title() string { return p.Name }`  for some reason `p` _is invalid receiver type *api.Playlist (type not defined in this package)_

Reply:
I'm not really sure about this but I believe item type needs to be declared in the same package based on Elm architecture that bubbles implements

Reply:
im not sure if this is really the issue, would it still throw an error if its not a pointer?

Reply:
Yes